src/updater.h

```diff
@@ -131,6 +132,7 @@ class MetadataUpdater {
 };
 
 // A class for all periodic updates of the metadata mapping.
+template<typename Clock = std::chrono::high_resolution_clock>
```

Comment:
(a) Why did you need to parameterize the whole class, rather than, e.g., providing a virtual `Now()` function that can be overridden by tests (or using an injected clock object in the constructor instead)?
(b) Don't you need to override the timer as well? Otherwise the timer will still lock for a particular period of time, whether your clock returns the new time earlier or not.